velox/functions/sparksql/DecimalVectorFunctions.cpp

```diff
@@ -0,0 +1,345 @@
+    aScaled = a * velox::DecimalUtil::kPowersOfTen[-deltaScale];
+  } else if (deltaScale > 0) {
+    bScaled = b * velox::DecimalUtil::kPowersOfTen[deltaScale];
+  }
+  if (aScaled == bScaled) {
+    return 0;
+  } else if (aScaled < bScaled) {
+    return -1;
+  } else {
+    return 1;
+  }
+}
+
+// Compare two decimals. Rescale one of them When they are of different scales.
+int32_t decimalCompare(
+    int128_t a,
+    uint8_t aPrecision,
+    uint8_t aScale,
+    int128_t b,
+    uint8_t bPrecision,
+    uint8_t bScale) {
+  int32_t deltaScale = aScale - bScale;
+  // Check if we need 256-bits after adjusting the scale.
+  bool need256 = (deltaScale < 0 &&
+                  aPrecision - deltaScale > LongDecimalType::kMaxPrecision) ||
+      (bPrecision + deltaScale > LongDecimalType::kMaxPrecision);
+  if (need256) {
+    return rescaleAndCompare<int256_t>(
+        static_cast<int256_t>(a), static_cast<int256_t>(b), deltaScale);
+  }
+  return rescaleAndCompare<int128_t>(a, b, deltaScale);
+}
+
+// GreaterThan decimal compare function.
+class Gt {
+ public:
+  inline static bool apply(
+      int128_t a,
+      uint8_t aPrecision,
+      uint8_t aScale,
+      int128_t b,
+      uint8_t bPrecision,
+      uint8_t bScale) {
+    return decimalCompare(a, aPrecision, aScale, b, bPrecision, bScale) > 0;
+  }
+};
+
+// GreaterThanOrEqual decimal compare function.
+class Gte {
+ public:
+  inline static bool apply(
+      int128_t a,
+      uint8_t aPrecision,
+      uint8_t aScale,
+      int128_t b,
+      uint8_t bPrecision,
+      uint8_t bScale) {
+    return decimalCompare(a, aPrecision, aScale, b, bPrecision, bScale) >= 0;
+  }
+};
+
+// LessThan decimal compare function.
+class Lt {
+ public:
+  inline static bool apply(
+      int128_t a,
+      uint8_t aPrecision,
+      uint8_t aScale,
+      int128_t b,
+      uint8_t bPrecision,
+      uint8_t bScale) {
+    return decimalCompare(a, aPrecision, aScale, b, bPrecision, bScale) < 0;
+  }
+};
+
+// LessThanOrEqual decimal compare function.
+class Lte {
+ public:
+  inline static bool apply(
+      int128_t a,
+      uint8_t aPrecision,
+      uint8_t aScale,
+      int128_t b,
+      uint8_t bPrecision,
+      uint8_t bScale) {
+    return decimalCompare(a, aPrecision, aScale, b, bPrecision, bScale) <= 0;
+  }
+};
+
+// Equal decimal compare function.
+class Eq {
+ public:
+  inline static bool apply(
+      int128_t a,
+      uint8_t aPrecision,
+      uint8_t aScale,
+      int128_t b,
+      uint8_t bPrecision,
+      uint8_t bScale) {
+    return decimalCompare(a, aPrecision, aScale, b, bPrecision, bScale) == 0;
+  }
+};
+
+// NotEqual decimal compare function.
+class Neq {
+ public:
+  inline static bool apply(
+      int128_t a,
+      uint8_t aPrecision,
+      uint8_t aScale,
+      int128_t b,
+      uint8_t bPrecision,
+      uint8_t bScale) {
+    return decimalCompare(a, aPrecision, aScale, b, bPrecision, bScale) != 0;
+  }
+};
+
+// Class for decimal compare operations.
```

Comment:
nit: This comment doesn't provide any additional information beyond what's provided by the class name. Let's remove or rewrite to provide more insight.

Reply:
Removed.